main.cpp

```diff
@@ -31,12 +31,11 @@ struct T
 
 struct Comparator                                //4
 {
-    T compare(T& a, T& b)//5
+    T* compare(T* a, T* b)//5
     {
-        if(a.value < b.value) return a;
-        if(a.value > b.value) return b;
-        T noneOfThem(0, "none of them");
-        return noneOfThem;
+        if(a->value < b->value) return a;
+        if(a->value > b->value) return b;
+        return nullptr;   
```

Comment:
you can change the function parameter type to references
you can't change the return type.  it must return T*
- [ ] revise